diff --git a/common/boot.c b/common/boot.c
index cef3d5e..7236a75 100644
--- a/common/boot.c
+++ b/common/boot.c
@@ -327,11 +327,13 @@ void bootsources_menu(struct bootentries *bootentries, int timeout)
 		menu_add_entry(bootentries->menu, &entry->me);
 	}
 
+#if 0
 	back_entry = xzalloc(sizeof(*back_entry));
 	back_entry->display = "back";
 	back_entry->type = MENU_ENTRY_NORMAL;
 	back_entry->non_re_ent = 1;
 	menu_add_entry(bootentries->menu, back_entry);
+#endif
 
 	if (timeout >= 0)
 		bootentries->menu->auto_select = timeout;
