diff --git a/common/efi/efi.c b/common/efi/efi.c
index 4b42f5d..b1d23c4 100644
--- a/common/efi/efi.c
+++ b/common/efi/efi.c
@@ -154,11 +154,11 @@ static struct efi_boot *efi_get_boot(int num)
 
 	ptr += (strlen(boot->description) + 1) * 2;
 
-	printf("description: %s\n", boot->description);
+	//printf("description: %s\n", boot->description);
 
 	boot->path = memdup(ptr, boot->file_path_len);
 
-	printf("path: %s\n", device_path_to_str(boot->path));
+	//printf("path: %s\n", device_path_to_str(boot->path));
 
 	return boot;
 }
