diff -aurN a/init b/init
--- a/init	2021-10-05 10:05:39.833895346 +0200
+++ b/init	2021-11-09 10:39:16.564294583 +0100
@@ -11,18 +11,6 @@
 #linux.bootargs.dyn.* will be cleared at the beginning of boot
 
 if [ -d /boot/boot/loader/entries ]; then
-    # signature verification
-    for file in /boot/boot/loader/entries/*.conf; do
-	verify_signature /env/pubkey.dtb ${file} ${file}.sig
-	if [ $? -ne 0 ]; then
-	    echo "Error verifying signature for ${file}."
-	    echo "!!! DENYING BOOT !!!"
-	    while true; do
-		sleep 1
-	    done
-	fi
-    done
-
     echo "Using BLSPEC on /boot"
     boot -m -t 5 /boot/boot
 fi
