diff --git a/commands/Kconfig b/commands/Kconfig
index ae2dc4b..d05a6aa 100644
--- a/commands/Kconfig
+++ b/commands/Kconfig
@@ -2127,6 +2127,14 @@ config CMD_SEED
 	help
 	  Seed the pseudo random number generator (PRNG)
 
+config CMD_SIGNATURE
+	tristate
+	prompt "signature"
+	select CRYPTO_RSA
+	select OFTREE
+	help
+	  Signature verification support
+
 # end Miscellaneous commands
 endmenu
 
diff --git a/commands/Makefile b/commands/Makefile
index 37486dc..525eff8 100644
--- a/commands/Makefile
+++ b/commands/Makefile
@@ -123,3 +123,4 @@ obj-$(CONFIG_CMD_SPD_DECODE)	+= spd_decode.o
 obj-$(CONFIG_CMD_MMC_EXTCSD)	+= mmc_extcsd.o
 obj-$(CONFIG_CMD_NAND_BITFLIP)	+= nand-bitflip.o
 obj-$(CONFIG_CMD_SEED)		+= seed.o
+obj-$(CONFIG_CMD_SIGNATURE)	+= signature.o
diff --git a/commands/signature.c b/commands/signature.c
new file mode 100644
index 0000000..8370fe9
--- /dev/null
+++ b/commands/signature.c
@@ -0,0 +1,108 @@
+#include <common.h>
+#include <command.h>
+#include <getopt.h>
+#include <errno.h>
+#include <rsa.h>
+#include <fs.h>
+#include <libfile.h>
+
+static int do_verify_signature(int argc, char *argv[])
+{
+	struct rsa_public_key key;
+        unsigned char *key_file = NULL , *file = NULL, *sig_file = NULL;
+        int opt, ret;
+	struct device_node *root_node, *key_node;
+        enum hash_algo algo = HASH_ALGO_SHA512;
+	void *sig_value = NULL, *hash_file = NULL, *hash = NULL;
+	size_t sig_len, hash_file_len, key_file_len;
+	struct digest *digest = NULL;
+
+        while ((opt = getopt(argc, argv, "a:s:")) > 0) {
+                switch(opt) {
+                case 'a':
+                        if(strcmp(optarg, "sha1") == 0)
+				algo = HASH_ALGO_SHA1;
+			else if(strcmp(optarg, "sha256") == 0)
+				algo = HASH_ALGO_SHA256;
+			else if(strcmp(optarg, "sha512") == 0)
+				algo = HASH_ALGO_SHA512;
+			else {
+				pr_err("Unknown algo %s\n", optarg);
+				return COMMAND_ERROR_USAGE;
+			}
+                        break;
+                default:
+                        return COMMAND_ERROR_USAGE;
+                }
+        }
+
+	key_file = read_file(argv[optind], &key_file_len);
+	if(!key_file) {
+		pr_err("Error reading keyfile %s\n", argv[optind]);
+		return COMMAND_ERROR_USAGE;
+	}
+	optind++;
+	file = argv[optind++];
+	sig_file = argv[optind++];
+
+	root_node = of_unflatten_dtb(key_file);
+	if(IS_ERR(root_node)) {
+		pr_err("Error reading key_file %d\n", (int)PTR_ERR(root_node));
+		return PTR_ERR(root_node);
+	}
+
+	key_node = of_find_node_by_path_from(root_node, "/signature/key-sig");
+	if(!key_node) {
+		pr_info("Failed to find key node\n");
+		ret = -ENOENT;
+		goto err_out;
+	}
+
+	if((ret = rsa_of_read_key(key_node, &key)))
+	{
+		pr_info("failed to read key in %s\n", key_node->full_name);
+                ret = -ENOENT;
+		goto err_out;
+	}
+
+	digest = digest_alloc_by_algo(algo);
+	BUG_ON(!digest);
+
+	digest_init(digest);
+	read_file_2(file, &hash_file_len, &hash_file, FILESIZE_MAX);
+	digest_update(digest, hash_file, hash_file_len);
+	hash = xzalloc(digest_length(digest));
+	digest_final(digest, hash);
+
+	ret = read_file_2(sig_file, &sig_len, &sig_value, FILESIZE_MAX);
+
+	ret = rsa_verify(&key, sig_value, sig_len, hash, algo);
+	if (ret) {
+		pr_info("file signature BAD\n");
+		ret = -EBADMSG;
+	} else {
+		pr_info("file signature OK\n");
+		ret = 0;
+	}
+
+err_out:
+	if(!digest)
+		digest_free(digest);
+	if(!hash)
+		free(hash);
+	if(!sig_value)
+		free(sig_value);
+	if(!key_file)
+		free(key_file);
+	if(!root_node)
+		of_delete_node(root_node);
+
+        return ret;
+}
+
+BAREBOX_CMD_START(verify_signature)
+	.cmd = do_verify_signature,
+	BAREBOX_CMD_DESC("Verify signature against known public key.")
+        BAREBOX_CMD_OPTS("[-a] keyfile file signature")
+	BAREBOX_CMD_GROUP(CMD_GRP_FILE)
+BAREBOX_CMD_END
